### ISSUE TITLE
Accessibility error in `Menu` component

### ISSUE THREAD

**Issue by loiswells97** [szhsin/react-menu#842]
React/React-dom version: `v3.2.0`

## Describe the bug

We are using your `Menu` component to create a context menu on our webpage and noticed that scanning for accessibility issues with the menu open raises an error in AXE devtools. This appears to be caused by the presence of a `<div>` inside the `<ul role='menu'>` element that contains the list of `menuitem`s. 

The HTML being rendered in our DOM looks like:
```html
<ul role="menu" aria-label="Menu" tabindex="-1" class="szh-menu szh-menu--state-open szh-menu--dir-bottom">
   <div tabindex="-1" style="position: absolute; left: 0px; top: 0px;" aria-hidden="true"></div>
   <li role="menuitem" tabindex="-1" class="szh-menu__item"></li>
   <li role="menuitem" tabindex="-1" class="szh-menu__item"></li>
</ul>
```
and the error received in AXE devtools is:
![Screenshot 2023-02-24 at 17 18 30](https://user-images.githubusercontent.com/88904316/221245168-b65a907e-6cda-498f-b50d-13bf24cb4457.png)

## To Reproduce

1. Render `Menu` component with `MenuItem` children
2. Run an AXE devtools scan with the menu open

## Potential Solution

It seems that including a `<div>` inside a `<ul>` is not valid HTML, and that AXE was expecting the all children of the `<ul role="menu">` element to have valid accessible roles such as `menuitem`.  One potential solution could be to make the `<div>` into an `<li>` and give it attributes `role='menuitem'` and `aria-hidden='true'`.

**Comment by szhsin** [comment]
@loiswells97  thank you for reporting the issue.

I guess adding `role='menuitem'` to an element which is not an actual menu item may cause more confusion for accessibility. Adding `role="presentation"` on a `li` could be the answer.

I don't have access to AXE devtools but did a quick check using the [w3 online checker](https://validator.w3.org/nu/#textarea), and the result had no warnings or errors.
```html
<!DOCTYPE html>
<html lang="">
  <head>
    <title>Test</title>
  </head>
  <body>
    <ul
      role="menu"
      aria-label="Open menu"
      tabindex="-1"
      class="szh-menu szh-menu--state-open szh-menu--dir-bottom"
      style="margin: 0px; position: absolute; left: 0px; top: 0px"
    >
      <li
        role="presentation"
        aria-hidden="true"
        tabindex="-1"
        style="position: absolute; left: 0px; top: 0px; display: block"
      ></li>
      <li role="menuitem" tabindex="-1" class="szh-menu__item">Item 1</li>
      <li role="menuitem" tabindex="-1" class="szh-menu__item">Item 2</li>
    </ul>
  </body>
</html>

```
If possible, could you verify the above markup in the AXE devtools for the result?

**Comment by loiswells97** [comment]
@szhsin thanks for your speedy reply! I've just tried your solution but sadly the AXE devtools are still throwing up the same error.

**Comment by szhsin** [comment]
Fixed in v3.5.2
